salt/modules/aptpkg.py

```diff
@@ -2995,3 +3000,38 @@ def list_downloaded(root=None, **kwargs):
                 ).isoformat(),
             }
     return ret
+
+
+def services_need_restart(**kwargs):
+    """
+    .. versionadded:: NEXT
+
+    List services that use files which have been changed by the
+    package manager. It might be needed to restart them.
+
+    Requires checkrestart from the debian-goodies package.
+
+    CLI Examples:
+
+    .. code-block:: bash
+
+        salt '*' pkg.services_need_restart
+    """
+    if not salt.utils.path.which_bin(["checkrestart"]):
+        raise CommandNotFoundError(
+            "'checkrestart' is needed. It is part of the 'debian-goodies' "
+            "package which can be installed from official repositories."
+        )
+
+    cmd = ["checkrestart", "--machine"]
```

Comment:
I think a `--package` is necessary to avoid certain false-positives. Or a way to take additional arguments.

Reply:
I think you're right with `--package` ([man page](https://manpages.debian.org/stretch/debian-goodies/checkrestart.1.en.html)).

Do you think it is needed to have other arguments?

Reply:
> Do you think it is needed to have other arguments?

I don't think so, but it's nice to be flexible if people have other ideas or new ones are added.

Reply:
I only worry that adding arbitrary arguments will move this from an implementation detail for gathering a list of services to something that will be used in a way that breaks the parsing done blow.
Parsing the output is already pretty brittle and not something I like doing, but it seems like the easiest way for achieving the goal for Debian systems (which I have to admit I have very little experience with).

Reply:
That's a fair point. Just adding `--package` would be sufficient.